File: Semana_2/Playwright/javascript/npm/curso/e2e/pages/separated/open-account.locators.js
class OpenAccountLocators {
  constructor(page) {
    this.page = page;
  }

  accountTypeSelect() {
    return this.page.getByRole('combobox').first();
  }

  fromAccountOptions() {
    return this.page.locator('#fromAccountId option');
  }

  openAccountButton() {
    return this.page.getByRole('button', { name: 'Open New Account' });
  }

  newAccountId() {
    return this.page.locator('#newAccountId');
  }
}

module.exports = { OpenAccountLocators };

File: Semana_2/Playwright/Java/maven/curso/e2e/src/test/java/com/curso/playwright/pages/separated/OpenAccountLocators.java
package com.curso.playwright.pages.separated;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.AriaRole;

public class OpenAccountLocators {
    private final Page page;

    public OpenAccountLocators(Page page) {
        this.page = page;
    }

    public Locator accountTypeSelect() {
        return page.getByRole(AriaRole.COMBOBOX).first();
    }

    public Locator fromAccountOptions() {
        return page.locator("#fromAccountId option");
    }

    public Locator openAccountButton() {
        return page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Open New Account"));
    }

    public Locator newAccountId() {
        return page.locator("#newAccountId");
    }
}
